brainbeats-backend/AuthConnection.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Graph;
+using Microsoft.Graph.Auth;
+using Microsoft.Identity.Client;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace brainbeats_backend {
+  public class AuthConnection {
+    public static AuthConnection Instance { get; set; }
+    private HttpClient httpClient { get; set; }
+    private GraphServiceClient graphClient { get; set; }
+    private string appId { get; set; }
+    private string tenantId { get; set; }
+
+    public static void Init(IConfiguration configuration) {
+      Instance = new AuthConnection(configuration);
+    }
+
+    private AuthConnection(IConfiguration configuration) {
+      appId = configuration["Auth:AppId"];
+      tenantId = configuration["Auth:TenantId"];
+      string clientSecret = configuration["Auth:ClientSecret"];
+
+      // Initialize new Microsoft Graph client
+      IConfidentialClientApplication confidentialClientApplication = ConfidentialClientApplicationBuilder
+          .Create(appId)
+          .WithTenantId(tenantId)
+          .WithClientSecret(clientSecret)
+          .Build();
+      ClientCredentialProvider authProvider = new ClientCredentialProvider(confidentialClientApplication);
+
+      graphClient = new GraphServiceClient(authProvider);
+
+      httpClient = new HttpClient();
+    }
+
+    public JwtSecurityToken ValidateToken(string token) {
+      if (!token.StartsWith("Bearer ")) {
+        throw new ArgumentException();
+      }
+
+      token = token.Split(" ")[1];
+
+      string stsDiscoveryEndpoint = "https://ucfbrainbeats.b2clogin.com/ucfbrainbeats.onmicrosoft.com/b2c_1_signupsignin/v2.0/.well-known/openid-configuration";
```

Comment:
This should be in a config file and not hardcoded (this url will probably change with Team 2 so lets not make their life harder)